### ISSUE TITLE
Not possible to save custom controller profiles

### ISSUE THREAD

**Issue by xordspar0** [flathub/org.libretro.RetroArch#256]
In #102, a user was confused about not being able to update the core info files; trying to do so results in a `EROFS (Read-only file system)` error. For data that is supposed to be read only like core and content databases, this is benign though a bit confusing.

However, joystick autoconfig is different. It's desirable to ship with some autoconfig so that common controllers are supported, but users may need to be able to configure their own controllers that aren't in the database. In particular, the Settings -> Inputs -> User 1 Binds -> User 1 Save Autoconfig button can never work. It always fails with "Error saving controller profile." Even if the user does copy an autoconfig file from a non-Flatpak install, that file will be ignored.

Is there any way to overlay .var/app/org.libretro.RetroArch/config/retroarch/autoconfig on top of the /app/share version? Or maybe RetroArch needs to grow support for a list of directories instead of just one.

> Correct, the .info files are shipped with the Flatpak, and are updated periodically through the Flatpak package.

> Would like to disable Online Updater menu items for components that are handled through > Flatpak:
> - Info files
> - Assets
> - Shaders
> - Joystick Autoconfig
> - Database

_Originally posted by @RobLoach in https://github.com/flathub/org.libretro.RetroArch/issues/102#issuecomment-461519964_

**Comment by RobLoach** [comment]
Would love to figure out a way to handle this, like merging the different paths or something.

**Comment by xordspar0** [comment]
> I don't know if you've already done this

Yeah, I changed the autoconfig directory to ~/.var/..., configured my controller, and submitted it to [retroarch-joypad-autoconfig](https://github.com/libretro/retroarch-joypad-autoconfig). So when that ships, I won't need this feature anymore. :slightly_smiling_face: 

> it doesn't make sense for asset files to be mixed with customizations made by users

Right, it seems like Retroarch wants everything to be writable; there is no distinction between assets and user configuration. Just download everything to a local directory and edit/add to it as you see fit. There are a few problems with doing it this way. For example, I assume that if you have a local autoconfig file that you made, and later a file with the same name is added to the official autoconfig repository, the next time you download the autoconfig files it will overwrite your local file?

So in short, I appreciate how the Flatpak version is trying to divide up user config and assets.

**Comment by davidhedlund** [comment]
* https://github.com/flathub/org.libretro.RetroArch/issues/295
